ui/src/views/Hooks/ListHooks/index.jsx

```diff
@@ -46,25 +48,80 @@ export default class ListHooks extends Component {
     });
   };
 
+  updateExpanded() {
+    this.query = qs.parse(this.props.location.search.slice(1));
+    const {
+      data: { hookGroups },
+    } = this.props;
+    const hookSearch = this.query.search;
+    const expandedHookGroups = hookGroups
+      ? hookGroups
+          .filter(
+            group =>
+              group.hooks.filter(hook => hook.hookId.includes(hookSearch))
+                .length !== 0
+          )
+          .map(group => group.hookGroupId)
+      : [];
+
+    this.setState({ expanded: expandedHookGroups, selected: [] });
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { expanded: [], selected: [] };
```

Comment:
I would write this as `this.state = this.updateExpanded()` but then you'd return an object in `updateExpanded` rather than setting state. The alternative is to calls `this.updateExpanded()` in `componentDidMount`.

Reply:
moved to component did mount

Reply:
This change introduced the query problem there is time between constructor and component did mount in which it is undefined, I can either move updateExpanded back to constructor or add this.query="" in constructor to give it initial value, what do you think?

Reply:
actually after removing the query from class it is no longer relevant